src/vehicle.cpp

```diff
@@ -4256,7 +4256,7 @@ int vehicle::get_z_change() const
 
 bool vehicle::would_prevent_flyable( const vpart_info &vpinfo ) const
 {
-    return is_flyable() && has_part( "ROTOR" ) && !vpinfo.has_flag( "SIMPLE_PART" );
```

Comment:
This should still check for the string `ROTOR` flag, this message shouldn't be triggering when it's a `SIMPLE_ROTOR`. The other change below this is potentially the same deal.